capsules/src/console.rs

```diff
@@ -204,15 +205,18 @@ impl<'a, U: UART> Client for Console<'a, U> {
         // Either print more from the AppSlice or send a callback to the
         // application.
         self.tx_buffer.replace(buffer);
-        self.in_progress.take().map(|appid| {
+        self.in_progress.get().map(|appid| {
+            self.in_progress.set(None);
```

Comment:
It could be a `MapCell`. This is i think the software engineering trade-off we talked about offline. Happy to change this to a MapCell as an example